src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,130 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [initialMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [movie, setMovie] = useState({ ...initialMovie });
+
+  const increment = () => {
+    setCount(prevCount => prevCount + 1);
+  };
+
+  const reset = () => {
+    setMovie(initialMovie);
+  };
+
+  const isNotValid =
+    movie.title.length === 0 ||
+    movie.imgUrl.length === 0 ||
+    movie.imdbUrl.length === 0 ||
+    movie.imdbId.length === 0;
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(movie);
+    increment();
+    reset();
+
+    if (isNotValid) {
+      // eslint-disable-next-line no-useless-return
+      return;
+    }
```

Comment:
looks like if not valid no action should be
```suggestion
    event.preventDefault();
    
    if (isNotValid) {
      return;
    }
    
    onAdd(movie);
    increment();
    reset();
```

Reply:
you apply validation check before you add a movie. the same as mentor provided